src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4028,18 +4034,16 @@ char* os::Linux::reserve_memory_special_huge_tlbfs(size_t bytes,
   char* small_start = aligned_start + large_bytes;
   size_t small_size = bytes - large_bytes;
   if (!large_committed) {
-    // Failed to commit large pages, so we need to unmap the
-    // reminder of the orinal reservation.
-    ::munmap(small_start, small_size);
+    // Failed to commit large pages, so we need to unmap the whole reservation.
+    ::munmap(aligned_start, bytes);
```

Comment:
I don't understand why this change is related to this ticket/PR, or needed at all.

Reply:
As mentioned above in case of large page commit failure we were partially un reserving the memory which left holes in the virtual address space , we thought of fixing this along with this patch if you suggest we can fix it under another bug id.

Reply:
The documentation around `mmap` with MAP_FIXED is not crystal clear on failure. In the case we return false from `commit_memory_special(...)` the commit over the reserved range failed and I'm pretty sure that when it does the virtual memory range reserved becomes available to the system again and might be reused by some other thread. So unmapping that range would break any such reservation. @stefank isn't this what we've had problems with in the past?

Reply:
Yes, that is my understanding.

Reply:
Yes, it was.

Reply:
I'm sorry, I don't fully follow your reasoning. Are you agreeing that this is not a problem?

To re-iterate what we have seen in the past: a reservation passing a NULL argument as the requested address is mapped on an address where a failed large page commit occurred (for this to happen, the previous reservation must have been release by the failure). This has lead us to require that the complete reservation is committed up front when using large pages, because we cannot risk loosing part of a reservation later in the execution. 

You can read more about this in the bug linked by @tstuefe and I think [this comments](https://bugs.openjdk.java.net/browse/JDK-8007074?focusedCommentId=13306490&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13306490) and the ones below explain the problem very well.

Reply:
Yes, it seem this may turn to be not-a-bug.